clusterman/autoscaler/autoscaler.py

```diff
@@ -141,6 +141,7 @@ def run(self, dry_run: bool = False, timestamp: Optional[arrow.Arrow] = None) ->
             logger.info('Autoscaling is currently paused; doing nothing')
             return
 
+        self.pool_manager.reload_state()
```

Comment:
nit: I'd add a comment here about this needing to happen before the pool signal is evaluated, especially we didn't catch this before.

Reply:
I did one better and added a test to catch this.